ocs_ci/ocs/machine.py

```diff
@@ -552,6 +552,199 @@ def create_custom_machineset(
             else:
                 raise ResourceNotFoundError("Machineset resource not found")
 
+    # check for vmware and IPI platform
+    elif config.ENV_DATA["platform"] == constants.VSPHERE_PLATFORM:
+        machinesets_obj = OCP(
+            kind=constants.MACHINESETS,
+            namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE,
+        )
+        for machine in machinesets_obj.get()["items"]:
+            # Get inputs from existing machineset config.
+            cls_id = (
+                machine.get("spec")
+                .get("selector")
+                .get("matchLabels")
+                .get("machine.openshift.io/cluster-api-cluster")
```

Comment:
why we are using this get approach?
If those data should be there and we need to use them, I would go with direct access from dict like:
machine.get["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]
I think it is more effective and will raise the direct keyError if we will get some data which do not have this key which will be more clear here than: ```
AttributeError: 'NoneType' object has no attribute 'get'```
Which doesn't tell you anything.

Instead of:
```
In [3]: data["spec"]["selector"]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-3-34fb83c900ad> in <module>
----> 1 data["spec"]["selector"]

KeyError: 'selector'
```
Is more precise on which data we are exactly missing.

Reply:
updated the code. Thanks for the comment @petr-balogh